src/Form/Type/AdminType.php

```diff
@@ -34,6 +35,30 @@
  */
 class AdminType extends AbstractType
 {
+    /**
+     * @var AdminHelper
+     */
+    private $adminHelper;
+
+    /**
+     * NEXT_MAJOR: remove the default null value.
+     */
+    public function __construct(?AdminHelper $adminHelper = null)
+    {
+        // NEXT_MAJOR: Remove this condition.
+        if (null === $adminHelper) {
+            @trigger_error(sprintf(
+                'Calling %s::%s without passing an %s as argument is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw an exception in 4.0.',
+                self::class,
+                __METHOD__,
```

Comment:
[`__METHOD__` already contains the classname](https://3v4l.org/rpIjZ), you might be confusing it with `__FUNCTION__`. It means this can be simplified.
```suggestion
                'Calling %s without passing an %s as argument is deprecated since sonata-project/admin-bundle 3.x'
                .' and will throw an exception in 4.0.',
                __METHOD__,
```